packages/release/release-sysctl.conf

```diff
@@ -2,8 +2,14 @@
 # Maximize console logging level for kernel printk messages
 kernel.printk = 8 4 1 7
 
-# Wait 30 seconds and then reboot
-kernel.panic = 30
+# Wait 10 seconds and then reboot
+kernel.panic = 10
+
+# Controls the kernel's behaviour when an oops or BUG is encountered
+kernel.panic_on_oops = 1
+
+# Overcommit handling mode - 1: Always overcommit
+vm.overcommit_memory = 1
```

Comment:
Do you have any information on why the [overcommit mode](https://www.kernel.org/doc/Documentation/vm/overcommit-accounting) would need to be changed from the default of 0 to 1?

I don't know if this would be appropriate for all variants.  In particular, this would be an unexpected change for the `aws-ecs-1` variant, as the AL2-based ECS-optimized AMIs inherit AL2's default setting of 0.

Reply:
> Do you have any information on why the [overcommit mode](https://www.kernel.org/doc/Documentation/vm/overcommit-accounting) would need to be changed from the default of 0 to 1?

Per https://github.com/kubernetes/kubernetes/issues/14935 it's done to disable the heuristics that could let a "best effort" container cause malloc to fail for a "guaranteed" container.

> I don't know if this would be appropriate for all variants.

If ECS doesn't have different QoS classes then it may not apply. I tend to want consistent behavior across variants here, though. The kernel's heuristics can introduce some surprising behaviors - running containers as root or not can give different results; applications can see unexpected malloc failures even if they're within their limits.

Allowing containers to exceed their reservation and killing them when they do is often much easier for operators to reason about.

Reply:
> If ECS doesn't have different QoS classes then it may not apply.

ECS does not have different QoS classes; there are no "best effort" or "guaranteed" containers.  Instead, ECS controls memory limits on containers via cgroups; every container has at least a memory limit or a memory reservation.

Reply:
@bcressey and I dug into this a bit more together.  The kernel behavior for `OVERCOMMIT_GUESS` (`0`, the default) [changed](https://github.com/torvalds/linux/commit/8c7829b04c523cdc732cb77f59f03320e09f3386) in Linux 5.2.  The new behavior is that allocations only fail if the requested virtual memory allocation exceeds the physical ram + swap space rather than using more complicated heuristics to guess.  This brings the behavior closer to `OVERCOMMIT_ALWAYS` (`1`), except that "wild allocations" will still fail.

Bottlerocket's kernel is 5.4, meaning that it includes this change.  With that said, my opinion is that we should leave the kernel configured with the default behavior (`OVERCOMMIT_GUESS`) in all variants; the QoS issues described in https://github.com/kubernetes/kubernetes/issues/14935 shouldn't happen in Bottlerocket.

Reply:
@samuelkarp found that my understanding is out-of-date, and the kernel documentation is no longer accurate since https://github.com/torvalds/linux/commit/8c7829b04c523cdc732cb77f59f03320e09f3386. There's no special treatment of root or processes with `CAP_SYS_ADMIN` since kernel 5.2+.

The difference is now:
 * `overcommit_memory = 0` only blocks requests for more memory than is physically present in ram + swap
 * `overcommit_memory = 1` allows all requests

I doubt the Kubernetes check is still relevant for newer kernels. It could be relaxed upstream to allow either `0` or `1` - but the application code would need to account for the running kernel version which is not recommended as a general practice.

The two paths I see are to patch our kubelets so they expect `0` instead of `1`, or ship a sysctl snippet for kubelet so that we can apply the expected value, and take advantage of the other checks for protected settings. I prefer the second approach.